nix/nixpkgs/default.nix

```diff
@@ -1,6 +1,17 @@
 let
   spec = builtins.fromJSON (builtins.readFile ./nixpkgs-src.json);
   rev = spec.rev;
+  sha256 = spec.sha256;
   url = "https://github.com/${spec.owner}/${spec.repo}/archive/${spec.rev}.tar.gz";
+
+  # fetchTarball version that is compatible between all the versions of Nix
+  fetch = { url, sha256 }@attrs:
+    let
+      inherit (builtins) lessThan nixVersion fetchTarball;
+    in
+    if lessThan nixVersion "1.12" then
+      fetchTarball { inherit url; }
```

Comment:
Unfortunately this means you don't have a fixed-output derivation. We contributed a bootstrapping Nix expression that does what you want, here: https://nixos.wiki/wiki/How_to_fetch_Nixpkgs_with_an_empty_NIX_PATH

Reply:
It's not ideal either because it now takes two sha256: one for the tarball and one for the unpacked file

Reply:
I think the inconvenience of providing two hashes outweighs the problems of a non-deterministic derivation for the less-than case, though.

Reply:
Does it prevent to pull from the binary cache? If not then I don't think it's an issue. If determinism is important, upgrade to 2.0.

Reply:
I have upgraded to Nix 2.0 but we don't have that luxury else-where at this time. Generally speaking, non-deterministic derivations (particularly ones we wrote for bootstrapping Nix) have bit us badly, I made this suggestion because I think the principled approach helps prevent annoying second-order problems that sometimes crop up from the expedient approach.

I also think it's fine to explicitly say you don't support anything less than Nix 2.0 to build the project.